lldb/source/Symbol/CoreDumpOptions.cpp

```diff
@@ -0,0 +1,45 @@
+//===-- CoreDumpOptions.cpp -------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "lldb/Symbol/CoreDumpOptions.h"
+
+using namespace lldb;
+using namespace lldb_private;
+
+void CoreDumpOptions::SetCoreDumpPluginName(const char * name) {
+  m_plugin_name = name;
```

Comment:
This will crash if `nullptr` is passed in. You need to check it and do the right thing:
```
if (name && name[0])
  m_plugin_name = name;
else:
  m_plugin_name = std::nullopt;
```
Add a test for this too please where you call `SetPluginName` with `NULL`